rules/aip0133/message_rules.go

```diff
@@ -0,0 +1,124 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0133
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/jhump/protoreflect/desc"
+	"github.com/jhump/protoreflect/desc/builder"
+)
+
+// The create request message should have parent field.
+var parentField = &lint.MessageRule{
+	Name: lint.NewRuleName("core", "0133", "request-message", "parent-field"),
+	URI:  "https://aip.dev/133#request-message",
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		// We only care about List- methods for the purpose of this rule;
+		// ignore everything else.
+		if !isCreateRequestMessage(m) {
+			return nil
+		}
+
+		// Rule check: Establish that a `parent` field is present.
+		parentField := m.FindFieldByName("parent")
+		if parentField == nil {
+			return []lint.Problem{{
+				Message:    fmt.Sprintf("Message %q has no `parent` field", m.GetName()),
+				Descriptor: m,
+			}}
+		}
+
+		// Rule check: Establish that the parent field is a string.
+		if parentField.GetType() != builder.FieldTypeString().GetType() {
+			return []lint.Problem{{
+				Message:    "`parent` field on create request message must be a string",
+				Descriptor: parentField,
+			}}
+		}
+
+		return nil
+	},
+}
+
+// The create request message should have resource field.
+var resourceField = &lint.MessageRule{
+	Name: lint.NewRuleName("core", "0133", "request-message", "resource-field"),
+	URI:  "https://aip.dev/133#request-message",
+	LintMessage: func(m *desc.MessageDescriptor) []lint.Problem {
+		// OnlyIf: isCreateRequestMessage
+		if !isCreateRequestMessage(m) {
+			return nil
+		}
+
+		resourceMsgName := getResourceMsgNameFromReq(m)
+
+		// The rule (resource field name must map to the POST body) is checked by AIP-0133 ("core::0133::http-body")
+		for _, fieldDesc := range m.GetFields() {
+			if msgDesc := fieldDesc.GetMessageType(); msgDesc != nil && msgDesc.GetName() == resourceMsgName {
+				return nil
+			}
+		}
+
+		// Rule check: Establish that a resource field must be included.
+		return []lint.Problem{{
+			Message:    fmt.Sprintf("Message %q has no %q type field", m.GetName(), resourceMsgName),
+			Descriptor: m,
+		}}
+	},
+}
+
+// The create request message should not have unrecognized fields.
+var unknownFields = &lint.MessageRule{
+	Name: lint.NewRuleName("core", "0133", "request-message", "unknown-fields"),
+	URI:  "https://aip.dev/133#request-message",
+	LintMessage: func(m *desc.MessageDescriptor) (problems []lint.Problem) {
+		// OnlyIf: isCreateRequestMessage
+		if !isCreateRequestMessage(m) {
+			return
+		}
+
+		resourceMsgName := getResourceMsgNameFromReq(m)
+
+		// Rule check: Establish that there are no unexpected fields.
+		allowedFields := map[string]*builder.FieldType{
+			"parent":     nil, // AIP-133
+			"request_id": nil, // AIP-155
+			fmt.Sprintf("%s_id", strings.ToLower(resourceMsgName)): nil,
```

Comment:
You probably need `strcase.SnakeCase` here too:

```go
strings.ToLower(strcase.SnakeCase(resourceMsgName))
```

Reply:
Done. Related unit testing is also updated.